### PR TITLE
doc: Add Quickstart Comet doc section

### LINKED ISSUE #57: Add documentation on how to use Comet
### What is the problem the feature request solves?

We should add some documentation in README (maybe a "Getting Started" section) on how to use Comet and what configurations are needed.

### Describe the potential solution

Add more documentation 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by comphead**
## Which issue does this PR close?

Closes #57 .

## What changes are included in this PR?
Quickstart Comet doc section

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,3 +58,45 @@ Linux, Apple OSX (Intel and M1)
...
+```
+cd arrow-datafusion-comet
+make release PROFILES="-Pspark-3.4"
+```
+
+### Run Spark with Comet enabled
+Make sure `SPARK_HOME` points to the same Spark version as Comet has built for.
+
+```
+spark-shell --jars spark/target/comet-spark-spark3.4_2.12-0.1.0-SNAPSHOT.jar \
```
Comment (sunchao) [inline README.md:83]: nit: `$SPARK_HOME/bin/spark-shell`?

File: README.md [hunk 2]
```diff
@@ -58,3 +58,45 @@ Linux, Apple OSX (Intel and M1)
...
+--conf spark.sql.extensions=org.apache.comet.CometSparkSessionExtensions \
+--conf spark.comet.enabled=true \
```
Comment (sunchao) [inline README.md:85]: we need a separate section to guide through shuffle related settings, but can be done later.

File: README.md [hunk 3]
```diff
@@ -58,3 +58,45 @@ Linux, Apple OSX (Intel and M1)
...
+
+### Clone repo
+```commandline
+git clone https://github.com/apache/arrow-datafusion-comet.git
+```
+### Specify the Spark version and build the Comet
+Spark 3.4 used for the example.
+```
+cd arrow-datafusion-comet
+make release PROFILES="-Pspark-3.4"
```
[the remark this answers is not in the document]
Comment (comphead) [reply README.md:76]: I'm not quite sure what can be improved here?

**Review by sunchao — APPROVED** [review]

**Comment by sunchao** [comment]
Merged, thanks!

File: README.md [hunk 4]
```diff
@@ -58,3 +58,49 @@ Linux, Apple OSX (Intel and M1)
...
+Create a test Parquet source
+```scala
+scala> (0 until 10).toDF("a").write.mode("overwrite").parquet("/tmp/test")
+```
+
+Query the data from the test source and check: 
+- INFO message shows the native Comet library has been initialized.
+- The query plan reflects Comet operators being used for this query instead of Spark ones
+```scala
+scala> spark.read.parquet("/tmp/test").createOrReplaceTempView("t1"); spark.sql("select * from t1 where a > 5").explain
```
Comment (viirya) [inline README.md:100]: It should be nicer to read if newline is added.

```
scala> spark.read.parquet("/tmp/test").createOrReplaceTempView("t1")
scala> spark.sql("select * from t1 where a > 5").explain
```

File: README.md [hunk 5]
```diff
@@ -58,3 +58,49 @@ Linux, Apple OSX (Intel and M1)
...
+INFO src/lib.rs: Comet native library initialized
+== Physical Plan ==
+        *(1) ColumnarToRow
+        +- CometFilter [a#14], (isnotnull(a#14) AND (a#14 > 5))
++- CometScan parquet [a#14] Batched: true, DataFilters: [isnotnull(a#14), (a#14 > 5)], Format: CometParquet, Location: InMemoryFileIndex(1 paths)[file:/tmp/test], PartitionFilters: [], PushedFilters: [IsNotNull(a), GreaterThan(a,5)], ReadSchema: struct<a:int>
```
Comment (viirya) [inline README.md:105]: And the explained plan string too. Right now it is a long string which is hard to read.

File: README.md (hunk 4 above)
Comment (comphead) [reply README.md:100]: I'll fix it.

File: README.md (hunk 5 above)
Comment (comphead) [reply README.md:105]: I'll break it into some newlines for easier reading

Comment (comphead) [reply README.md:105]: Filed https://github.com/apache/arrow-datafusion-comet/pull/128

**Pull request merged**